OasisDEX/oasis-borrow
# New sidebar component

# New sidebar component

Covers:
https://app.shortcut.com/oazo-apps/story/4216/create-new-configurable-sidebar-component
https://app.shortcut.com/oazo-apps/story/4222/create-reset-button-on-new-opening-vault-component
  
## Changes 👷‍♀️

Created new component `SidebarSection` that allows to display vault forms in new designs. This PR is only about wrapper component, not content inside of it. Scrolling inside of a content is part of another story.

![image](https://user-images.githubusercontent.com/16230404/167878007-9b04ef9d-4667-415f-b9ab-d90e4b32fc6b.png)
  
## How to test 🧪

Since this is only the component and not the implementation, you will need to put it into the code on local environment. The best way to test it is in `features/borrow/open/containers/OpenVaultView.tsx` file, include `SidebarSection` component next to (or instead of) current sidebar, which is `VaultFormContainer`.

`SidebarSection` has a lot of options and it would be good to test them all and see how they interact with each other. Below are couple examples of preconfigured component with some dummy data.

**Example 1**
```
<SidebarSection
  title="Lorem ipsum"
  dropdown={[
    {
      label: 'Manage collateral (ETH)',
      shortLabel: 'ETH',
      icon: getToken('ETH').iconCircle,
      panel: 'eth',
    },
    {
      label: 'Manage DAI',
      shortLabel: 'DAI',
      icon: getToken('DAI').iconCircle,
      panel: 'dai',
    },
    {
      label: 'Switch to multiply',
      action: () => {
        alert('Dropdown action clicked')
      },
    },
  ]}
  content={[
    {
      panel: 'eth',
      content: <Box>Displays ETH panel open</Box>,
    },
    {
      panel: 'dai',
      content: <Box>Displays DAI panel open</Box>,
    },
  ]}
  primaryButton={{
    label: 'Primary action',
    action: () => {
      alert('Primary action clicked')
    },
  }}
  textButton={{
    label: 'Additional action',
    action: () => {
      alert('Additional action clicked')
    },
  }}
/>
```
**Example 2**
```
<SidebarSection
  title="Lorem ipsum"
  headerButton={{
    label: 'Header action',
    icon: 'refresh',
    action: () => {
      alert('Header action clicked')
    },
  }}
  content={<Box>Single panel sidebar</Box>}
  primaryButton={{
    label: 'Primary action',
    isLoading: true,
    disabled: true,
    action: () => {
      alert('Primary action clicked')
    },
  }}
  progress={{
    text: 'Some stuff in progress',
    txHash: '',
    etherscan: '',
  }}
/>
```
